### PR TITLE
[#560] Remove separation between debug and default mode

### LINKED ISSUE #560: Get rid of `debug` mode
Now that logging is configurable, there is no real need for a `debug` mode. I would get rid of the whole concept and always include redbug as a dependency. If we decide to do so, we should update all clients and docs that refer to a `debug` mode.

### PR DESCRIPTION
**Pull request by robertoaloi**
### Description

* Remove `debug` mode, keeping the alias
* Enable logging by default

Fixes #560 .

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -29,21 +31,14 @@
...
-                 }
-               , { escript_emu_args
-                 , "%%! -connect_all false -erlang_ls logging_enabled true \n"
-                 }
-                 %% For some reason we need to explicitly include redbug,
-                 %% even if it already is a dependency.
-               , {escript_incl_apps, [redbug]}
-               ]
-             }
+%% Keeping the debug profile as an alias, since many clients were
```
Comment (jfacorro) [inline rebar.config]: In what ways were clients relying on the `debug` profile? Because of the path to the escript?

File: src/erlang_ls.erl [hunk 2]
```diff
@@ -111,7 +111,6 @@ set(transport, Name) ->
 set(port, Port) ->
   application:set_env(?APP, port, Port);
 set(log_dir, Dir) ->
-  application:set_env(?APP, logging_enabled, true),
   application:set_env(?APP, log_dir, Dir);
 set(log_level, Level) ->
   application:set_env(?APP, log_level, Level);
```
Comment (jfacorro) [inline src/erlang_ls.erl]: I just noticed that the rewrite of parsing the options using `getopt` removed the enabling of logging when a `log_level` is provided. I guess it's saner to just enable it in one place or have it enabled by default.

File: rebar.config (hunk 1 above)
Comment (robertoaloi) [reply rebar.config]: Yes, exactly.

File: src/erlang_ls.erl (hunk 2 above)
Comment (robertoaloi) [reply src/erlang_ls.erl]: Sure, feel free to raise an issue.

File: src/erlang_ls.erl [hunk 3]
```diff
@@ -125,7 +124,7 @@ set(port_old, Port) ->
 
 -spec lager_config() -> ok.
 lager_config() ->
-  LoggingEnabled = application:get_env(?APP, logging_enabled, false),
+  LoggingEnabled = application:get_env(?APP, logging_enabled, true),
```
Comment (jfacorro) [inline src/erlang_ls.erl]: The value in `erlang_ls.app.src` is `false`, though. Do we want this to be `true` or `false` by default?

Comment (alanz) [reply src/erlang_ls.erl]: I would prefer it to be true, in general. It makes it easier to ask people for extra info when they report problems, to not have to first take them through a process of getting logging enabled.

Comment (robertoaloi) [reply src/erlang_ls.erl]: That was my intention, but missed the `.app.src` entry. Will follow-up.

Comment (jfacorro) [reply src/erlang_ls.erl]: I used to be of a different opinion, but I now think that logging enabled is a good default.

**Pull request merged**